Taken from Fedora cvs.

gunze_calib is declared as int[4], so referring to idx #4 is 
obviously a bad idea :).

--- gpm-1.20.1/src/mice.c.lr	2004-02-15 09:19:48.000000000 +0100
+++ gpm-1.20.1/src/mice.c	2004-02-15 09:22:18.000000000 +0100
@@ -2112,7 +2112,7 @@
       for (i=0; i<4; i++)
          if (gunze_calib[i] & ~1023) calibok = 0;
       if (gunze_calib[0] == gunze_calib[2]) calibok = 0;
-      if (gunze_calib[1] == gunze_calib[4]) calibok = 0;
+      if (gunze_calib[1] == gunze_calib[3]) calibok = 0;
       fclose(f);
    }
    if (!calibok) {
